extensions/mongodb-client/deployment/src/test/java/io/quarkus/mongodb/NamedMongoClientConfigTest.java

```diff
@@ -0,0 +1,43 @@
+package io.quarkus.mongodb;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import javax.inject.Inject;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.mongodb.runtime.MongoClientConnection;
+import io.quarkus.test.QuarkusUnitTest;
+
+public class NamedMongoClientConfigTest extends MongoWithReplicasTestBase {
+
+    @RegisterExtension
+    static final QuarkusUnitTest config = new QuarkusUnitTest()
+            .withConfigurationResource("application-named-mongoclient.properties");
+
+    @Inject
+    @MongoClientConnection("cluster1")
+    ReactiveMongoClient client;
+
+    @Inject
+    @MongoClientConnection("cluster2")
+    ReactiveMongoClient client2;
```

Comment:
I think we should also have a test for `MongoClient`, not just `ReactiveMongoClient`